numba/core/typing/new_builtins.py

```diff
@@ -0,0 +1,1178 @@
+import itertools
+
+import numpy as np
+import operator
+
+from numba.core import types, errors, config
+from numba import prange
+from numba.parfors.parfor import internal_prange
+
+from numba.core.typing.templates import (AttributeTemplate, ConcreteTemplate,
+                                         AbstractTemplate, infer_global, infer,
+                                         infer_getattr, signature,
+                                         bound_function, make_callable_template)
+
+
+from numba.core.extending import (
+    typeof_impl, type_callable, models, register_model, make_attribute_wrapper,
+    )
+
+
+@infer_global(print)
+class Print(AbstractTemplate):
+    def generic(self, args, kws):
+        for a in args:
+            sig = self.context.resolve_function_type("print_item", (a,), {})
+            if sig is None:
+                raise TypeError("Type %s is not printable." % a)
+            assert sig.return_type is types.none
+        return signature(types.none, *args)
+
+@infer
+class PrintItem(AbstractTemplate):
+    key = "print_item"
+
+    def generic(self, args, kws):
+        arg, = args
+        return signature(types.none, *args)
+
+
+@infer_global(abs)
+class Abs(ConcreteTemplate):
+    int_cases = [signature(ty, ty) for ty in sorted(types.np_signed_domain)]
+    uint_cases = [signature(ty, ty) for ty in sorted(types.np_unsigned_domain)]
+    real_cases = [signature(ty, ty) for ty in sorted(types.np_real_domain)]
+    complex_cases = [signature(ty.underlying_float, ty)
+                     for ty in sorted(types.np_complex_domain)]
+    cases = int_cases + uint_cases +  real_cases + complex_cases
+
+
+@infer_global(slice)
+class Slice(ConcreteTemplate):
+    cases = [
+        signature(types.slice2_type, types.np_intp),
+        signature(types.slice2_type, types.none),
+        signature(types.slice2_type, types.none, types.none),
+        signature(types.slice2_type, types.none, types.np_intp),
+        signature(types.slice2_type, types.np_intp, types.none),
+        signature(types.slice2_type, types.np_intp, types.np_intp),
+        signature(types.slice3_type, types.np_intp, types.np_intp, types.np_intp),
+        signature(types.slice3_type, types.none, types.np_intp, types.np_intp),
+        signature(types.slice3_type, types.np_intp, types.none, types.np_intp),
+        signature(types.slice3_type, types.np_intp, types.np_intp, types.none),
+        signature(types.slice3_type, types.np_intp, types.none, types.none),
+        signature(types.slice3_type, types.none, types.np_intp, types.none),
+        signature(types.slice3_type, types.none, types.none, types.np_intp),
+        signature(types.slice3_type, types.none, types.none, types.none),
```

Comment:
This should probably all refer to Python `types.py_int` as it this this which the `slice` constructor consumes?